src/features/session_replay/aggregate/index.js

```diff
@@ -0,0 +1,292 @@
+    /** Set once an error has been detected on the page.  Is cleared after a harvest */
+    this.errorNoticed = false
+    /** The "mode" to record in.  Defaults to "OFF" until flags and sampling are checked. See "MODE" constant. */
+    this.mode = MODE.OFF
+    /** Set once the feature has been "aborted" to prevent other side-effects from continuing */
+    this.blocked = false
+
+    /** Payload metadata -- Should indicate that the payload being sent is the first of a session */
+    this.isFirstChunk = false
+    /** Payload metadata -- Should indicate that the payload being sent has a full DOM snapshot. */
+    this.hasSnapshot = false
+    /** Payload metadata -- Should indicate that the payload being sent contains an error. */
+    this.hasError = false
+
+    /** A value which increments with every new mutation node reported. Resets after a harvest is sent */
+    this.payloadBytesEstimation = 0
+
+    /** The method to stop recording. This defaults to a noop, but is overwritten once the recording library is imported and initialized */
+    this.stopRecording = () => { /* no-op until set by rrweb initializer */ }
+
+    // The SessionEntity class can emit a message indicating the session was cleared and reset (expiry, inactivity). This feature must abort and never resume if that occurs.
+    this.ee.on('session-reset', () => {
+      this.abort()
+    })
+
+    // The SessionEntity class can emit a message indicating the session was paused (visibility change). This feature must stop recording if that occurs.
+    this.ee.on(SESSION_EVENTS.PAUSE, () => { this.stopRecording() })
+    // The SessionEntity class can emit a message indicating the session was resumed (visibility change). This feature must start running again (if already running) if that occurs.
+    this.ee.on(SESSION_EVENTS.RESUME, () => {
+      if (!this.initialized || this.mode === MODE.OFF) return
+      this.startRecording()
+      this.takeFullSnapshot()
+    })
+
+    // Bespoke logic for new endpoint.  This will change as downstream dependencies become solidified.
+    this.scheduler = new HarvestScheduler('blob', {
+      onFinished: this.onHarvestFinished.bind(this),
+      retryDelay: this.harvestTimeSeconds,
+      getPayload: this.prepareHarvest.bind(this),
+      // TODO -- this stuff needs a better way to be handled
+      includeBaseParams: false,
+      customUrl: 'https://vortex-alb.stg-single-tooth.cell.us.nr-data.net/blob',
+      raw: true,
+      gzip: true
+    }, this)
+
+    // Wait for an error to be reported.  This currently is wrapped around the "Error" feature.  This is a feature-feature dependency.
+    // This was to ensure that all errors, including those on the page before load and those handled with "noticeError" are accounted for. Needs evalulation
+    registerHandler('err', (e) => {
+      this.hasError = true
+      // run once
+      if (this.errorNoticed) return
+      // if the error was noticed AFTER the recorder was imported....
```

Comment:
If the error was noticed _before_ `initializeRecording` imports rrweb's record, the mode should still switch to FULL. One such case can happen due to race condition since that function is async.